x-pack/plugin/ent-search/qa/full-cluster-restart/javaRestTest/java/org/elasticsearch/entsearch/FullClusterRestartIT.java

```diff
@@ -0,0 +1,115 @@
+
+    @ClassRule
+    public static final ElasticsearchCluster cluster = ElasticsearchCluster.local()
+        .distribution(DistributionType.DEFAULT)
+        .version(getOldClusterTestVersion())
+        .nodes(2)
+        .setting("xpack.security.enabled", "false")
+        .module("x-pack-ent-search")
+        .build();
+
+    public FullClusterRestartIT(@Name("cluster") FullClusterRestartUpgradeStatus upgradeStatus) {
+        super(upgradeStatus);
+    }
+
+    @Override
+    protected ElasticsearchCluster getUpgradeCluster() {
+        return cluster;
+    }
+
+    public void testBehavioralAnalyticsDataRetention() throws Exception {
+
+        String legacyAnalyticsCollectionName = "oldstuff";
+        String newAnalyticsCollectionName = "newstuff";
+
+        if (isRunningAgainstOldCluster()) {
+            // Create an analytics collection
+            Request legacyPutRequest = new Request("PUT", "_application/analytics/" + legacyAnalyticsCollectionName);
+            assertOK(client().performRequest(legacyPutRequest));
+
+            // Validate that ILM lifecycle is in place
+            assertBusy(() -> testLegacyDataRetentionPolicy(legacyAnalyticsCollectionName));
+        } else {
+            // Create a new analytics collection
+            Request putRequest = new Request("PUT", "_application/analytics/" + newAnalyticsCollectionName);
+            assertOK(client().performRequest(putRequest));
+
+            // Validate that NO ILM lifecycle is in place and we are using DLM instead.
+            assertBusy(() -> testDlmDataRetentionPolicy(newAnalyticsCollectionName));
+
+            // Validate that the existing analytics collection created with an older version is still using ILM
+            assertBusy(() -> testLegacyDataRetentionPolicy(legacyAnalyticsCollectionName));
+        }
+    }
+
+    private void testLegacyDataRetentionPolicy(String analyticsCollectionName) throws IOException {
+        String dataStreamName = "behavioral_analytics-events-" + analyticsCollectionName;
+        Request getDataStreamRequest = new Request("GET", "_data_stream/" + dataStreamName);
+        Response response = client().performRequest(getDataStreamRequest);
+        assertOK(response);
+        ObjectPath dataStream = ObjectPath.createFromResponse(response);
+        String pathToIlmPolicy = "data_streams.0.ilm_policy";
+        assertNotNull(dataStream.evaluate(pathToIlmPolicy));
+        assertEquals(EVENT_DATA_STREAM_LEGACY_ILM_POLICY_NAME, dataStream.evaluate(pathToIlmPolicy));
+
+        Request policyRequest = new Request("GET", "_ilm/policy/" + EVENT_DATA_STREAM_LEGACY_ILM_POLICY_NAME);
+        ObjectPath policy = ObjectPath.createFromResponse(client().performRequest(policyRequest));
+        assertNotNull(policy.evaluate(EVENT_DATA_STREAM_LEGACY_ILM_POLICY_NAME));
+    }
+
+    private void testDlmDataRetentionPolicy(String analyticsCollectionName) throws IOException {
```

Comment:
I usually try to not use `test` prefix for method that are not tests.
I would recommend to rename the method into `assertDslDataRetentionPolicy` (which make sense since used into a `assertBusy`